gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+
+        # Yaml file now exists with none value.
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Test bad
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            # Corrupt file and user doesn't choose to delete and re-create 
+            make_yaml('key value')
+            configuration_tests.load_yaml_value('yaml.yaml', 'key')
+
+        if os.path.isfile('yaml.yaml'):
+            os.remove('yaml.yaml')  
+
+
+    @mock.patch('gslab_scons.configuration_tests.is_unix')
+    @mock.patch('gslab_scons.configuration_tests.convert_packages_argument')
+    @mock.patch('gslab_scons.configuration_tests.subprocess.check_output')
+    def test_check_stata_packages(self, mock_check_output, mock_package_argument, 
+                                  mock_is_unix):
+
+        def check_output_side_effect(*args, **kwargs):
+            command = args[0].split("which ")[0]
+            if not re.search("statamp", command):
+                raise subprocess.CalledProcessError(1, args[0])  
+
+            with open('stata.log', 'wb') as f:
+                if not re.search('yaml', args[0]):
+                    package = args[0].split("which ")[1]
+                    f.write("command %s not found" % package)
+
+        def is_unix_side_effect_true():
+            return True
+ 
+        def is_unix_side_effect_false():
+            return False
+
+        mock_check_output.side_effect     = check_output_side_effect
+        mock_package_argument.side_effect = lambda x: ['yaml']
+        mock_is_unix.side_effect          = is_unix_side_effect_true
+
+        with mock.patch('gslab_scons.configuration_tests.sys.platform', 'win32'):
+            # Good tests
+
+            # Unix
+            configuration_tests.check_stata_packages("statamp", "yaml")
+
+            # Windows
+            mock_is_unix.side_effect = is_unix_side_effect_false
+            configuration_tests.check_stata_packages("statamp", "yaml")
+
+            # No packages
+            mock_package_argument.side_effect = lambda x: []
+            configuration_tests.check_stata_packages("statamp", [])
+
+            # Bad Tests
+            with self.assertRaises(ex_classes.PrerequisiteError):
+                # Bad package
+                mock_package_argument.side_effect = lambda x: ['bad_package']
+                configuration_tests.check_stata_packages("statamp", "bad_package")
+            
+            with self.assertRaises(ex_classes.PrerequisiteError):
+                # Bad executable
+                mock_package_argument.side_effect = lambda x: ['yaml']
+                configuration_tests.check_stata_packages("bad_executable", "yaml")
+
+        # More bad tests
+        with self.assertRaises(ex_classes.PrerequisiteError): 
+            mock.patch('gslab_scons.configuration_tests.sys.platform', 'Unknown')
```

Comment:
Can you patch like this? I thought you have to use `with mock.patch()` or something similar.